[16.0][ADD/MIG] l10n_ch_qr_bill_scan

michelerusti opened the pull request:
This PR replaces #698 
Due to the complicated history of this module, I'd like to clarify that this is both a migration and a module addition.

The module presented in #553 for 13.0 (then superseded by #718) was never merged but nonetheless migrated to different versions ([14.0](https://github.com/OCA/l10n-switzerland/pull/671), [15.0](https://github.com/OCA/l10n-switzerland/pull/671), even backported to the [10.0](https://github.com/OCA/l10n-switzerland/pull/561)

None of the PRs mentioned above have been merged. I hope I didn't miss any relevant context, but this is another attempt to bring this module to the main branch, making it a module addition—thus the branch name.

yvaucher committed:
[13.0][ADD] l10n_ch_qr_bill_scan

A new module to scan QR-bill or to import PDF files

yvaucher committed:
Fix currency setting in unit test

yvaucher committed:
Fix debug message of decoded data

i-vyshnevska committed:
[IMP] improve UI

yankinmax committed:
[IMP] l10n_ch_qr_bill_scan: add UserError after QR-code Swiss cross found but cant be readable

yankinmax committed:
[IMP] l10n_ch_qr_bill_scan: add test on PDF import with wrong QR

matiasperalta1 committed:
[IMP] l10n_ch_qr_bill_scan: black, isort, prettier

ajaniszewska-dev committed:
[IMP] : black, isort, prettier

ajaniszewska-dev committed:
[MIG] l10n_ch_qr_bill_scan: Migration to 15.0

ajaniszewska-dev committed:
fixup! [MIG] l10n_ch_qr_bill_scan: Migration to 15.0

Commit:
[IMP] l10n_ch_qr_bill_scan: pre-commit stuff

Commit:
[ADD] l10n_ch_qr_bill_scan: Add module to 16.0

stevech091 reviewed (approved):
Functional test ok.

michelerusti commented:
@OCA/local-switzerland-maintainers could you please re-open this PR? 
We'll provide technical reviews

michelerusti committed:
FIX test

TDu commented on l10n_ch_qr_bill_scan/wizard/account_invoice_import.py:
```diff
@@ -33,7 +33,7 @@ class AccountInvoiceImport(models.TransientModel):
 
     state = fields.Selection(
-        selection_add=[("select-partner", "Select partner")],
+        selection=[("select-partner", "Select partner")],
```
The implementation of the `account_invoice_import` module has been improved by the author in https://github.com/OCA/edi/pull/753
I think we should follow the same logic here and not add this `state` field that has been removed.

michelerusti committed:
[IMP] l10n_ch_qr_bill_scan: align with new account_invoice_import architecture

- Remove state and partner-related fields from the wizard, as they are now handled by the base module via a banner/wizard on the created invoice.
- Update import_invoices to handle the new multi-file attachment flow by creating an attachment from the invoice_scan field.
- Refactor parse_pdf_invoice and parse_qrbill to match new signatures and simplify the parsing flow.
- Improve plain text QR data detection to handle leading whitespace and different line endings.
- Update tests to use new res.partner configuration fields and verify import_partner_data on invoices.
